Extension/src/LanguageServer/extension.ts

```diff
@@ -83,6 +100,119 @@ export function activate(activationEventOccurred: boolean): void {
     }
 }
 
+/**
+ * Generate tasks to build the current file based on the user's detected compilers, the user's compilerPath setting, and the current file's extension.
+ */
+async function getBuildTasks(): Promise<vscode.Task[]> {
+    const activeEditor: vscode.TextEditor = vscode.window.activeTextEditor;
+    if (!activeEditor) {
+        return [];
+    }
+
+    // Don't offer tasks for header files.
+    const activeFileExt: string = path.extname(activeEditor.document.fileName).toLowerCase();
+    const isHeader: boolean = [".hpp", ".hh", ".hxx", ".h", ""].some(ext => activeFileExt === ext);
+    if (isHeader) {
+        return [];
+    }
+
+    // Don't offer tasks if the active file's extension is not a recognized C/C++ extension.
+    const activeFileIsCpp: boolean = [".cpp", ".cc", ".cxx", ".mm", ".ino", ".inl"].some(ext => activeFileExt === ext);
```

Comment:
if this is case-sensitive, consider dealing with `.C` (which is c++, by convention).  Perhaps we should offer both in this case.